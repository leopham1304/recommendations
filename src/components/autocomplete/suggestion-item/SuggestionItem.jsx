import React, { memo } from "react";
import PropTypes from "prop-types";
import { useHandleSelect } from "./SuggestionItem.utils";

export const SuggestionItem = ({ onSelect, suggestion }) => {
  const handleSelect = useHandleSelect({ onSelect, suggestion });
  return <li onClick={handleSelect}>{suggestion}</li>;
};

SuggestionItem.propTypes = {
  onSelect: PropTypes.func.isRequired,
  suggestion: PropTypes.string.isRequired,
};

export default memo(SuggestionItem);
