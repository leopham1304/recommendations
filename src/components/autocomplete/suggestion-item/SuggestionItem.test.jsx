import React from "react";
import SuggestionItem from "./SuggestionItem";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

const onSelectMock = jest.fn();

describe("SuggestionItem", () => {
  const wrapper = mount(
    <SuggestionItem onSelect={onSelectMock} suggestion="Test" />
  );

  it("should render the SuggestionItem Component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should bind onSelect properly", () => {
    wrapper.find("li").simulate("click");
    expect(onSelectMock).toBeCalled();
  });
});
