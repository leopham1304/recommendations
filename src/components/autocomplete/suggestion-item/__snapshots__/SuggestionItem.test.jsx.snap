// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionItem should render the SuggestionItem Component correctly 1`] = `
<Memo(SuggestionItem)
  onSelect={[MockFunction]}
  suggestion="Test"
>
  <li
    onClick={[Function]}
  >
    Test
  </li>
</Memo(SuggestionItem)>
`;
